### PR TITLE
fix: allow dots in username

### LINKED ISSUE #11581: [CAL-2539] feat: ability to use one "." inside a username
allowing "." would mean we can auto convert: [leo.giovanetti@acme.com](mailto:leo.giovanetti@acme.com) to [acme.cal.com/leo.giovanetti](http://acme.cal.com/leo.giovanetti) for Enterprise/Org users

## requirements

* username cannot start or end with "."
* no limits of "."
* cannot have two dots ".."
* ~~**maybe?** we need to introduce a banlist for file extensions (.exe, .jpg, .svg, etc.)~~

see google:

CleanShot 2023-09-28 at 13 23 21@2x

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2539](https://linear.app/calcom/issue/CAL-2539/feat-ability-to-use-one-inside-a-username)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/11581

https://github.com/calcom/website/pull/624

How to test ?

- Go to profile settings and try changing the username to 'john.doe'.
OR
- create an org 
- invite a user with email (john.doe@acme.com) and then open the invite link on another browser to verify it's username is 'john.doe' and not 'john-doe' (like we did earlier)

Screenshot 2023-10-05 at 4 49 18 PM

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: packages/lib/slugify.test.ts
Comment (sean-brydon) [inline packages/lib/slugify.test.ts:35]: Do we want to allow usernames like this? 🤔

Comment (Udit-takkar) [reply packages/lib/slugify.test.ts:35]: Screenshot 2023-10-06 at 12 33 25 PM

It was mentioned in the issue requirements.

Comment (PeerRich) [reply packages/lib/slugify.test.ts:35]: > Do we want to allow usernames like this? 🤔

yes, since it's actually the official spec of email usernames

**Review by PeerRich — DISMISSED** [review]
looks pretty good to me but waiting for @zomars as final Auth-pro for merge

**Review by zomars — DISMISSED** [review]
Overall looks good to me. Will block until some concerns of mine are addressed and properly tested.

- Can we create an actual dotted username in the e2e seeder?
- Can we verify avatar URLs will work as expected?
- Can we test dynamic bookings work with dotted usernames?
- Can we test cal.com/dotted.user rewrites correctly to app.cal.com/dotter.user ?

I'm just being extra careful here. There are a lot of things that could go wrong since suglify is used everywhere.

**Comment by Udit-takkar** [comment]
1) i am able to seed this username. (yarn db-seed)

Screenshot 2023-10-13 at 10 14 22 PM

2) Avatar Url
Screenshot 2023-10-13 at 9 16 23 PM

3) Dynamic bookings

Screenshot 2023-10-13 at 9 16 47 PM

4) How to test this locally?

**Comment by PeerRich** [comment]
> 4. How to test this locally?

i think this would work on cal.dev/u.d.i.t once its merged to `main` 

not sure how to test locally

**Review by zomars — DISMISSED** [review]
By testing I've meant automated e2e test. Manual testing here helps but it won't help us prove/disprove a regression. Let's add dotted username to normal booking and dynamic booking e2e tests before merging.

**Comment by hariombalhara** [comment]
Tested following
- Booking a user like `team.pro `works
- Dynamic group booking like `team.pro+teamfree` works
- Team Slug still doesn't allow dot, maybe we want to allow that too as a followup.
- Event Slug is now accepting dot, not sure if it's expected. Booking an event-type like /team/marketing/a.b.c works
Screenshot 2023-10-20 at 12 24 53

**Review by hariombalhara — APPROVED** [review]
LGTM!! Great work @Udit-takkar !!

**Comment by zomars** [comment]
Heads up this is breaking `https://app.cal.com/embed.js`

**Pull request merged**
